inc/saiobject.h

```diff
@@ -249,6 +249,56 @@ sai_status_t sai_query_stats_capability(
         _In_ sai_object_type_t object_type,
         _Inout_ sai_stat_capability_list_t *stats_capability);
 
+/**
+ * @brief Bulk objects get statistics.
+ *
+ * @param[in] switch_id SAI Switch object id
+ * @param[in] object_type Object type
+ * @param[in] object_count Number of objects to get the stats
+ * @param[in] object_key List of object keys
+ * @param[in] number_of_counters Number of counters in the array
+ * @param[in] counter_ids Specifies the array of counter ids
+ * @param[in] mode Statistics mode
+ * @param[inout] object_statuses Array of status for each object. Length of the array should be object_count. Should be looked only if API return is not SAI_STATUS_SUCCESS.
+ * @param[out] counters Array of resulting counter values.
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_bulk_object_get_stats_fn)(
+        _In_ sai_object_id_t switch_id,
+        _In_ sai_object_type_t object_type,
+        _In_ uint32_t object_count,
+        _In_ const sai_object_key_t *object_key,
+        _In_ uint32_t number_of_counters,
+        _In_ const sai_stat_id_t *counter_ids,
+        _In_ sai_stats_mode_t mode,
+        _Inout_ sai_status_t *object_statuses,
+        _Out_ uint64_t *counters);
```

Comment:
should this be double array "uint64_t **counters"?? since we have N objects keys, and for each object key we are getting M counter_ids

Reply:
> should this be double array "uint64_t **counters"?? since we have N objects keys, and for each object key we are getting M counter_ids

This indeed can be presented as double array. But I saw at other places that convention is just to pass a pointer to a block of buffer and that is cast'ed as needed. Hence kept the same here.

Reply:
I have updated comments; how the counter buffer is populated.

Reply:
@kcudnik - please review? Thanks.

Reply:
i think in all other places apis like that, when multiple objects are used, then array of arrays is used, like bulk get, bulk create